invisible_cities/io/mcinfo_io.py

```diff
@@ -330,32 +456,68 @@ def read_mcparticles_df(h5in    : tb.file.File,
     return parts
 
 
-def load_mcsensor_response(file_name: str,
-                           event_range=(0, int(1e9))) -> Mapping[int, Mapping[int, Waveform]]:
-
-    with tb.open_file(file_name, mode='r') as h5in:
-        return read_mcsns_response(h5in, event_range)
-
-
-def get_sensor_binning(file_name : str) -> Tuple:
+def get_sensor_binning(file_name : str) -> pd.DataFrame:
     """
     Looks in the configuration table of the
     input file and extracts the binning used
     for both types of sensitive detector.
     """
-    config   = pd.read_hdf(file_name, 'MC/configuration')
-    bins     = config[config.param_key.str.contains('time_binning')]
-    pmt_bin  = bins.param_value[bins.param_key.str.contains('Pmt')].iloc[0]
-    pmt_bin  = float(pmt_bin.split()[0]) * units_dict[pmt_bin.split()[1]]
-    sipm_bin = bins.param_value[bins.param_key.str.contains('SiPM')].iloc[0]
-    sipm_bin = float(sipm_bin.split()[0]) * units_dict[sipm_bin.split()[1]]
+    if is_oldformat_file(file_name):
+        return sensor_binning_old(file_name)
+    else:
+        return sensor_binning_new(file_name)
+
+
+def sensor_binning_old(file_name : str) -> pd.DataFrame:
+    # Is a pre-Feb 2020 MC file
+    config = pd.read_hdf(file_name, 'MC/configuration').set_index('param_key')
+    bins         = config[config.index.str.contains('binning')].copy()
+    bins.columns = ['bin_width']
+    bins.index   = bins.index.rename('sns_name')
+    ## Protect against badly written old files
+    bins         = bins[bins.index.str.contains('Geom')]
+    ##
+    bins.index   = bins.index.str.split('/', expand=True).levels[2]
+    bins.bin_width = bins.bin_width.str.split(expand=True).apply(
+        lambda x: float(x[0]) * getattr(units, x[1]), axis=1)
+    return bins
+
+
+def sensor_binning_new(file_name : str) -> pd.DataFrame:
+    config = pd.read_hdf(file_name, 'MC/configuration').set_index('param_key')
+    sns_names    = pd.read_hdf(file_name, 'MC/sns_positions').sensor_name
+    if len(sns_names) == 0:
+        return pd.DataFrame(index=['sns_name'], columns=['bin_width'])
+    sns_binnings = [nm + '_binning' for nm in sns_names.unique()]
+    bins         = config.loc[sns_binnings].copy()
+    bins.columns = ['bin_width']
+    bins.index   = bins.index.rename('sns_name')
+    bins.index   = bins.index.str.strip('_binning')
+    bins.bin_width = bins.bin_width.str.split(expand=True).apply(
+        lambda x: float(x[0]) * getattr(units, x[1]), axis=1)
+    return bins
+
+
+def get_sensor_types(file_name : str) -> pd.DataFrame:
```

Comment:
I am confused between get, load and read names, can't we make a more systematic naming?